lib/plugins/static.js

```diff
@@ -55,29 +56,48 @@ function serveStatic(options) {
             res.handledGzip();
         }
 
-        var fstream = fs.createReadStream(file + (isGzip ? '.gz' : ''));
         var maxAge = opts.maxAge === undefined ? 3600 : opts.maxAge;
-        fstream.once('open', function (fd) {
-            res.cache({maxAge: maxAge});
-            res.set('Content-Length', stats.size);
-            res.set('Content-Type', mime.lookup(file));
-            res.set('Last-Modified', stats.mtime);
-
-            if (opts.charSet) {
-                var type = res.getHeader('Content-Type') +
-                    '; charset=' + opts.charSet;
-                res.setHeader('Content-Type', type);
-            }
+        res.cache({maxAge: maxAge});
+        res.set('Content-Length', stats.size);
+        res.set('Content-Type', mime.lookup(file));
+        res.set('Last-Modified', stats.mtime);
+
+        if (opts.charSet) {
+            var type = res.getHeader('Content-Type') +
+                '; charset=' + opts.charSet;
+            res.setHeader('Content-Type', type);
+        }
+
+        if (opts.etag) {
+            res.set('ETag', opts.etag(stats, opts));
+        }
+
+        var requestTime;
+        var modifiedTime;
 
-            if (opts.etag) {
-                res.set('ETag', opts.etag(stats, opts));
+        try {
+            if (opts.checkIfModified) {
+                requestTime = Date.parse(req.header('if-modified-since')) || 0;
+                modifiedTime = Date.parse(stats.mtime);
             }
-            res.writeHead(200);
-            fstream.pipe(res);
-            fstream.once('end', function () {
-                next(false);
+        } catch (e) {
+            opts.checkIfModified = false;
```

Comment:
This error shouldn't be swallowed. What is the expected error? I believe `Date.parse` doesn't throw an error like JSON.parse, rather it returns NaN if it encounters invalid data. This try catch shouldn't be necessary.